rpc/src/rpc.rs

```diff
@@ -1289,6 +1289,16 @@ impl JsonRpcRequestProcessor {
             .unwrap()
             .highest_super_majority_root();
 
+        if commitment.is_finalized() {
```

Comment:
makes me wonder whether an `RpcContextConfig` ought to be invalid if `min_context_slot` is set with `commitment` other than `Finalized`?

Reply:
Generally in rpc? Or in this method specifically?

Reply:
generally, right? 'cause any other commitment level could theoretically query two slots on different forks

Reply:
I see what you're saying, but I don't think minimum context slot promises anything about the data returned; just that the node has progressed up to a particular point in chain time. Commitment is still the key to turn if those guarantees are important to the caller.

Reply:
maybe less about a promise and more that `min_context_slot` is unintuitively(?) ignored for other commitment levels

Reply:
It's not ignored for other commitment levels, though:
https://github.com/anza-xyz/agave/blob/f8bb98b5f4e7710a5ea75a44c0c0f6eddfb09dd7/rpc/src/rpc.rs#L217-L231